.github/workflows/basic.yml

```diff
@@ -35,20 +40,41 @@ jobs:
         sudo apt-get -y install zlib1g-dev libssl-dev uuid-dev libcurl4-openssl-dev
         mkdir $CMAKE_BUILD_DIR
 
+    - name: Install Prereqs - MacOS and OpenSSL Version 1.1
+      if: startsWith(matrix.os, 'macOS') && (matrix.openssl-version == 1.1)
+      shell: bash
+      run: |
+        echo "Uninstalling OpenSSL3"
+        brew uninstall --ignore-dependencies openssl@3
+        brew cleanup -s openssl@3
+        brew cleanup --prune-prefix
+
+
     - name: Install Prereqs - MacOS
       if: startsWith(matrix.os, 'macOS')
       shell: bash
       run: |
         brew list cmake &>/dev/null || brew install cmake
         brew list lcov &>/dev/null || brew install lcov
-        brew list openssl@1.1 &>/dev/null || brew install openssl@1.1
+        brew list openssl@${{matrix.openssl-version}} &>/dev/null || brew install openssl@${{matrix.openssl-version}}
         mkdir $CMAKE_BUILD_DIR
 
     - name: Build
       shell: bash
       working-directory: ${{env.CMAKE_BUILD_DIR}}
       run: |
+        export OPENSSL_VERSION=${{matrix.openssl-version}}
+        if [[ !  -z  ${OPENSSL_VERSION} ]]; then
+          crypto_lib_path=`brew list openssl@${{matrix.openssl-version}} | grep "libcrypto.dylib$"`
```

Comment:
This `brew list` portion is Mac specific...but this section of Actions could be run even in the ubuntu case right? Maybe we're relying on the fact that `openssl-version` is only specified in the Mac case right now...but we should move the Mac portion to the part above that is gated by `if: startsWith(matrix.os, 'macOS')` and pass it to this portion, maybe using env vars.

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable